<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from shelf.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="shelf">
  <link name="world"/>
  <joint name="fixed" type="fixed">
    <parent link="world"/>
    <child link="shelf"/>
    <origin rpy="0 0 3.14159" xyz="0.6 0 0"/>
  </joint>
  <link name="shelf"/>
  <link name="door">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="5.8"/>
      <inertia ixx="0.434826" ixy="0" ixz="0" iyy="0.391536656" iyz="0" izz="0.434826"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0.15 0"/>
      <geometry>
        <box size="0.02 0.3 0.9"/>
      </geometry>
      <material name="door_material">
        <color rgba="1.0 1.0 1.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.15 0"/>
      <geometry>
        <box size="0.02 0.3 0.9"/>
      </geometry>
    </collision>
  </link>
  <link name="door_hinge"/>
  <joint name="door_hinge_joint" type="fixed">
    <parent link="shelf"/>
    <child link="door_hinge"/>
    <dynamics damping="2.0" friction="4.0"/>
    <origin xyz="0 -0.3 0.45"/>
  </joint>
  <joint name="articulation_joint" type="revolute">
    <parent link="door_hinge"/>
    <child link="door"/>
    <dynamics damping="2.0" friction="4.0"/>
    <origin xyz="0 0 0"/>
    <axis xyz="0 0 -1"/>
    <limit effort="100" lower="0" upper="1.57" velocity="10.0"/>
  </joint>
  <link name="handle">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.17"/>
      </geometry>
      <material name="handle_material">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.17"/>
      </geometry>
    </collision>
  </link>
  <link name="handle_support_left">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.036 0.01 0.01"/>
      </geometry>
      <material name="handle_material">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.036 0.01 0.01"/>
      </geometry>
    </collision>
  </link>
  <link name="handle_support_right">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.036 0.01 0.01"/>
      </geometry>
      <material name="handle_material">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.036 0.01 0.01"/>
      </geometry>
    </collision>
  </link>
  <joint name="handle_joint" type="fixed">
    <parent link="door"/>
    <child link="handle"/>
    <origin rpy="0 0 0" xyz="0.046 0.25 0"/>
    <!-- height handle + half the width door, 0.30 minus some offset -->
  </joint>
  <joint name="handle_support_left_joint" type="fixed">
    <parent link="handle"/>
    <child link="handle_support_left"/>
    <origin rpy="0 0 0" xyz="-0.018 0 0.08"/>
  </joint>
  <joint name="handle_support_right_joint" type="fixed">
    <parent link="handle"/>
    <child link="handle_support_right"/>
    <origin rpy="0 0 0" xyz="-0.018 0 -0.08"/>
  </joint>
</robot>

